File: umove-supervisor-ui/src/app/model/vehicle.ts
import {VehicleType} from "./vehicle-type";

export class Vehicle{
  public id: string;
  public zoneid: string;
  public insuranceNo: string;
  public status: string;
  public registrationNo: string;
  public type: VehicleType;
  public lastServiceDate: Date;
  public vehiclePurchased: Date;
}

File: zone-service/src/main/java/in/stackroute/umove/zoneservice/model/Vehicle.java
package in.stackroute.umove.zoneservice.model;

import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document
@Getter
@Setter
@Data
@AllArgsConstructor
@NoArgsConstructor

public class Vehicle {

    private String id;
    private String zoneid;
    private String registrationNo;
    private String insuranceNo;
    private VehicleStatus status;
    private VehicleType type;
    private Date time=new Date();
    private Date lastServiceDate;
    private Date vehiclePurchased;
    private String chassisNumber;

    public Vehicle(String id, String zoneid, String registrationNo) {
        this.id = id;
        this.zoneid = zoneid;
        this.registrationNo = registrationNo;
    }
}
